text/0074-stateful-multisig-pallet.md

```diff
@@ -0,0 +1,293 @@
+
+### Requirements
+
+Basic requirements for the Stateful Multisig are:
+
+* The ability to have concrete and permanent (unless deleted) multisig accounts in the state.
+* The ability to add/remove owners from an existing multisig account by the multisig itself.
+* The ability to change the threshold of an existing multisig account by the multisig itself.
+* The ability to delete an existing multisig account by the multisig itself.
+* The ability to cancel an existing proposal by the multisig itself.
+* Owners of multisig account can start a proposal on behalf of the multisig account which will be dispatched after getting enough approvals.
+* Owners of multisig account can approve/revoke a proposal while still exists.
+
+### Use Cases
+
+* Corporate Governance:
+In a corporate setting, multisig accounts can be employed for decision-making processes. For example, a company may require the approval of multiple executives to initiate significant financial transactions.
+
+* Joint Accounts:
+Multisig accounts can be used for joint accounts where multiple individuals need to authorize transactions. This is particularly useful in family finances or shared business accounts.
+
+* Decentralized Autonomous Organizations (DAOs):
+DAOs can utilize multisig accounts to ensure that decisions are made collectively. Multiple key holders can be required to approve changes to the organization's rules or the allocation of funds.
+
+and much more...
+
+## Stakeholders
+
+* Polkadot holders
+* Polkadot developers
+
+## Explanation
+
+I've created the stateful multisig pallet during my studies in Polkadot Blockchain Academy under supervision from @shawntabrizi and @ank4n. After that, I've enhanced it to be fully functional and this is a draft [PR#3300](https://github.com/paritytech/polkadot-sdk/pull/3300) in polkadot-sdk. I'll list all the details and design decisions in the following sections.
+
+Let's start with a sequence diagram to illustrate the main operations of the Stateful Multisig.
+
+![multisig operations](https://github.com/asoliman92/RFCs/assets/2677789/4f2e8972-f3b8-4250-b75f-1e4788b35752)
+
+Notes on above diagram:
+
+* It's a 3 step process to execute a proposal. (Start Proposal --> Approvals --> Execute Proposal)
+* `Execute` is an explicit extrinsic for a simpler API. It can be optimized to be executed automatically after getting enough approvals.
+* Any user can create a multisig account and they don't need to be part of it. (Alice in the diagram)
+* A proposal is any extrinsic including control extrinsics (e.g. add/remove owner, change threshold,..etc).
+* Any multisig account owner can start a proposal on behalf of the multisig account. (Bob in the diagram)
+* Any multisig account owener can execute proposal if it's approved by enough owners. (Dave in the diagram)
+
+### State Transition Functions
+
+All functions have detailed rustdoc in [PR#3300](https://github.com/paritytech/polkadot-sdk/pull/3300). Here is a brief overview of the functions:
```

Comment:
the RPC is the spec, not the code, so we should have all the technical details here
in this case, we need arguments and detailed descriptions such as origin requirement, events, deposits etc

Reply:
Thanks! I'm adding them now 👍